File: engine/model/src/main/scala/tools/packet/guild/ShowPlayerRanks.scala
package tools.packet.guild

import client.database.data.GlobalUserRank
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowPlayerRanks(private var _npcId: Int, private var _ranks: java.util.List[GlobalUserRank]) extends PacketInput {
  def npcId: Int = _npcId

  def ranks: java.util.List[GlobalUserRank] = _ranks

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/ShowPlayerRanks.java
package tools.packet.guild;

import java.util.List;

import client.database.data.GlobalUserRank;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowPlayerRanks(Integer npcId, List<GlobalUserRank> ranks) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}